pkg/app/api/httpapi/httpapimetrics/handler.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpapimetrics
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	codeLabel   = "code"
+	methodLabel = "method"
+	pathLabel   = "path"
+)
+
+var (
+	requestCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_requests_total",
+			Help: "Total number of HTTP requests.",
+		},
+		[]string{codeLabel, methodLabel, pathLabel},
+	)
+
+	durationHistgram = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "http_request_duration_milliseconds",
+			Help:    "Histogram of request latencies in milliseconds.",
+			Buckets: prometheus.ExponentialBuckets(0.0001, 500, 5),
```

Comment:
(1, 5, 7) looks better. It produces [1 5 25 125 625 3125 15625].
It will be good if our HTTP requests will be handled under 125ms.

Reply:
Right. What the factor is 5 looks good. I thought to observe correctly, start should be around 0.1us because the root `/` doesn't take a time. But what we have to observe is a kind of duration spikes. So let me apply your suggestion

Reply:
Well how about (0.1, 5, 7)? It will be [0.1, 0.5, 2.5, 12.5, 62.5, 312.5, 1562.5]